include/boost/geometry/views/detail/boundary_view/implementation.hpp

```diff
@@ -0,0 +1,400 @@
+    {}
+
+    // for begin
+    polygon_rings_iterator(Polygon& polygon)
+        : m_polygon(boost::addressof(polygon))
+        , m_index(0)
+    {}
+
+    // for end
+    polygon_rings_iterator(Polygon& polygon, bool)
+        : m_polygon(boost::addressof(polygon))
+        , m_index(num_rings(polygon))
+    {}
+
+    template
+    <
+        typename OtherPolygon,
+        typename OtherValue,
+        typename OtherReference,
+        typename OtherDifference
+    >
+    polygon_rings_iterator(polygon_rings_iterator
+                           <
+                               OtherPolygon,
+                               OtherValue,
+                               OtherReference,
+                               OtherDifference
+                           > const& other)
+        : m_polygon(other.m_polygon)
+        , m_index(other.m_index)
+    {
+        static const bool is_convertible
+            = boost::is_convertible<OtherPolygon, Polygon>::value;
+
+        BOOST_MPL_ASSERT_MSG((is_convertible),
+                             NOT_CONVERTIBLE,
+                             (types<OtherPolygon>));
+    }
+
+private:
+    friend class boost::iterator_core_access;
+
+    template
+    <
+        typename OtherPolygon,
+        typename OtherValue,
+        typename OtherReference,
+        typename OtherDifference
+    >
+    friend class polygon_rings_iterator;
+
+
+    static inline std::size_t num_rings(Polygon const& polygon)
+    {
+        return geometry::num_interior_rings(polygon) + 1;
+    }
+
+    inline Reference dereference() const
+    {
+        if (m_index == 0)
+        {
+            return exterior_ring(*m_polygon);
+        }
+        return range::at(interior_rings(*m_polygon), m_index - 1);
+    }
+
+    template
+    <
+        typename OtherPolygon,
+        typename OtherValue,
+        typename OtherReference,
+        typename OtherDifference
+    >
+    inline bool equal(polygon_rings_iterator
+                      <
+                          OtherPolygon,
+                          OtherValue,
+                          OtherReference,
+                          OtherDifference
+                      > const& other) const
+    {
+        return m_polygon == other.m_polygon && m_index == other.m_index;
+    }
+
+    inline void increment()
+    {
+        ++m_index;
+    }
+
+    inline void decrement()
+    {
+        --m_index;
+    }
+
+    template
+    <
+        typename OtherPolygon,
+        typename OtherValue,
+        typename OtherReference,
+        typename OtherDifference
+    >
+    inline Difference distance_to(polygon_rings_iterator
+                                  <
+                                      OtherPolygon,
+                                      OtherValue,
+                                      OtherReference,
+                                      OtherDifference
+                                  > const& other) const
+    {
+        return other.m_index - m_index;
```

Comment:
Here the result of subtraction of two unsigned values is converted into signed value. Will this work on all platforms?